compiler/rustc_codegen_ssa/src/mir/naked_asm.rs

```diff
@@ -211,8 +211,10 @@ fn prefix_and_suffix<'tcx>(
             writeln!(begin, ".pushsection {section},\"ax\", {progbits}").unwrap();
             writeln!(begin, ".balign {align}").unwrap();
             write_linkage(&mut begin).unwrap();
-            if let Visibility::Hidden = item_data.visibility {
-                writeln!(begin, ".hidden {asm_name}").unwrap();
+            match item_data.visibility {
+                Visibility::Default => {}
+                Visibility::Protected => writeln!(begin, ".protected {asm_name}").unwrap(),
+                Visibility::Hidden => writeln!(begin, ".hidden {asm_name}").unwrap(),
```

Comment:
I think we discussed documenting https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/mono/enum.Visibility.html a while ago, now might be a good time for that? (you had some ideas for that I believe)?

Also I don't think `.hidden` and `.protected` are ever hit by tests, is there any way we could reach them?

Reply:
> I think we discussed documenting https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/mono/enum.Visibility.html a while ago, now might be a good time for that? (you had some ideas for that I believe)?

Will open a separate PR for that.

> Also I don't think .hidden and .protected are ever hit by tests, is there any way we could reach them?

You can set the visibility for all symbols using `-Zdefault-visibility`.

Reply:
yeah that's fine, we just want to make sure that naked functions behave like standard functions